pkg/apis/garden/validation/validation.go

```diff
@@ -1340,11 +1405,33 @@ func validateCloud(cloud garden.Cloud, fldPath *field.Path) field.ErrorList {
 			if workerNames[worker.Name] {
 				allErrs = append(allErrs, field.Duplicate(idxPath, worker.Name))
 			}
+			if worker.Kubelet != nil && worker.Kubelet.MaxPods != nil && *worker.Kubelet.MaxPods > maxPod {
+				maxPod = *worker.Kubelet.MaxPods
+			}
 			workerNames[worker.Name] = true
 		}
 
 	}
 
+	allErrs = append(allErrs, ValidateNodeCIDRMaskWithMaxPod(maxPod, *kubernetes.KubeControllerManager.NodeCIDRMaskSize)...)
+
+	return allErrs
+}
+
+// ValidateNodeCIDRMaskWithMaxPod validates if the Pod Network has enough ip addresses (configured via the NodeCIDRMask on the kube controller manager) to support the highest max pod setting on the shoot
+func ValidateNodeCIDRMaskWithMaxPod(maxPod int32, nodeCIDRMaskSize int) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if maxPod == 0 {
+		// default maxPod setting on kubelet
+		maxPod = 110
```

Comment:
I would not default that here, but rather [here](https://github.com/gardener/gardener/pull/1299/files#diff-eaaa98b650b977d0655f31a7fc8de71fR1030). This function should only validate what it gets.

Reply:
If we set the MaxPod in the location suggested by you (before we calculate the actually set maxPod's) - the maxPod minimum is always 110. However it can be desired and configured (globally & on the worker pools) that the maxPod minimum is < 110  resulting in a smaller NodeCIDRMask

Reply:
OK, good point. Then default it before calling this function, please. I still think such function should not do defaulting.